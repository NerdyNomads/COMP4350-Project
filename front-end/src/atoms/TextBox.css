.TextBox {
    width: 355px;
    height: 213px;
    background-color: rgb(255, 255, 255);    
    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.23);
    border-radius: 15px;
    margin: 10px;
    min-width: 355px;
    min-height: 213px;
}

.TextBox:hover {     
    background-color: rgb(250, 250, 250); ;    
}

.text {
    margin: 45px 45px;
    margin-bottom: 70px;
    min-height: 45px;
    max-height: 45px;
    cursor: pointer;
}

.textfield {
    height: 15px;
    width: 250px;
    align-self: flex-end;
    font-size: 12px;
    color: #808080;
    font-style: italic;
    border: none;
}

.card-footer {
    margin: 13px 20px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
}

.source {
    user-select:none;
    cursor:pointer;
}

.divider {
    border: 4px solid #F3F3F3;
    user-select:none;
    background-color: #F3F3F3;
}

.delete {
    cursor: pointer;
    user-select:none;
}