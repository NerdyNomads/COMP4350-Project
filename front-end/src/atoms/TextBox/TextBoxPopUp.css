
.TextBoxPopUp-background{

  background-color: rgba(0, 0, 0, 0.692);
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 50;

  display: flex;
  flex-direction: column;
  align-items: center;

}


.TextBoxPopUp{
    min-width: 300px;
    max-width: 1022px;
    max-height: 693px;
    height:fit-content;

    background: #FFFFFF;
    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.23);
    border-radius: 15px;

    display: flex;
    flex-direction: column;
    padding: 15px;
    margin: 0px 50px;

    position: fixed;
    z-index: 51;
    top:10%;
}

.TextBoxPopUp-exit{
    align-self: flex-end;
    margin-right: 5px;
    cursor: pointer;
}

.TextBoxPopUp-source{
    margin-left: 20px;
    display: flex;
    flex-direction: row;
    cursor: pointer;
}

.TextBoxPopUp-source-icon{
    margin-right: 8px;

}
.TextBoxPopUp-source-text{
    font-size: 12px;
    line-height: 150%;
    color: #808080;
    align-self: center;
    font-style: italic;
}

.TextBoxPopUp-source-text-a{
  all: unset;
}

.TextBoxPopUp-text{
    text-align: left;
    margin: 20px;
    margin-top: 5px;
    overflow: auto;
}


/* custom scrollbar */
::-webkit-scrollbar {
    width: 20px;
  }
  
  ::-webkit-scrollbar-track {
    background-color: transparent;
  }
  
  ::-webkit-scrollbar-thumb {
    background-color: #d6dee1;
    border-radius: 20px;
    border: 6px solid transparent;
    background-clip: content-box;
  }
  
  ::-webkit-scrollbar-thumb:hover {
    background-color: #a8bbbf;
  }
  /* custom scrollbar ends */