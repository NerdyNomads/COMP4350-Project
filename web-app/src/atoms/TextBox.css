.TextBox {
    width: 355px;
    height: 213px;
    border: 1px solid white;
    background-color: rgb(255, 255, 255);    
    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.23);
    border-radius: 15px;
    margin: 10px;
}

.text {
    margin: 45px 45px;
    margin-bottom: 70px;
    max-height: 45px;
    cursor: pointer;
}

.card-footer {
    margin: 13px 20px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.source {
    user-select:none;
}

.divider {
    border: 4px solid #F3F3F3;
    user-select:none;
}

.delete {
    cursor: pointer;
    user-select:none;
}