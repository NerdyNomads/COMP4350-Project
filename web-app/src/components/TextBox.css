.text-box {
	background-color: white;
	width: 55vw;
	margin-left: 5vh;
	margin-top: 5px;
	color: black;
	border-radius: 2vh;
	font-size: 20px;
}
