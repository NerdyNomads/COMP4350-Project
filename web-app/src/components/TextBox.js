import React from "react";

class TextBox extends React.Component {
    render() {
        return (
            <div className= "text-box">
                <b>My Inspiration</b>
                <div>It does not matter how fast you go as long as you do not stop.</div>
            </div>
        )
    }
}

export default TextBox;