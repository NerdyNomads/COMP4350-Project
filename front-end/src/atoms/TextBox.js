import React from "react";
import PropTypes from "prop-types";

import { ChainIcon, TrashCanIcon } from "./icons";
import "./TextBox.css";


const MAX_CHARACTERS = 142;


function TextBox({text, source}) {

  const formatText = (t) => {
    if (t.length > MAX_CHARACTERS) {
      t = t.slice(0, MAX_CHARACTERS-4);
      t = `"${t}..."`;
    }

    return t;
  };

  const handleCardClick = () => {
    alert(`Temporary: ${text}`);
  };

  const handleCopyLink = () => {
    const temp = document.createElement("textarea");
    temp.value = source;
    document.body.appendChild(temp);
    temp.select();
    document.execCommand("copy");
    document.body.removeChild(temp);

    alert(`Link copied: ${source}`);

    text = document.getElementById("link");
    if (text.childElementCount == 0) {
      temp.className = "textarea";
      text.appendChild(temp);
    }
  };

  const handleDelete = () => {
    alert("Temporary: delete");
  };


  return (
    <div className="TextBox">
      <div className="text" onClick={handleCardClick}>
        {formatText(text)}
      </div>
      <div className="divider"/>
      <div className="card-footer">
        <div className="source" target="_blank" rel="noreferrer" onClick={handleCopyLink}>
          <ChainIcon />
        </div>
        <div id="link">
        </div>
        <div onClick={handleDelete} className="delete">
          <TrashCanIcon/>
        </div>
      </div>
    </div>
  );
}

TextBox.propTypes = {
  text: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired

};

export default TextBox;