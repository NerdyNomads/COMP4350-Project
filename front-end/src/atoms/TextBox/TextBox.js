import React, { useState } from "react";
import PropTypes from "prop-types";

import { trimLongText } from "../../util/util";
import { ChainIcon, TrashCanIcon } from "../icons";
import TextBoxPopUp from "./TextBoxPopUp";
import "./TextBox.css";

import { deleteText } from "../../util/requests";

const MAX_CHARACTERS = 142;

function TextBox({textItem, onDelete}) {
  const [showTextPopUp, setShowTextPopUp] = useState(false);

  const handleDelete = async (id) => {
    await deleteText(id);
    onDelete(id);
  };

  const formatText = (t) => {
    if (t.length > MAX_CHARACTERS) {
      t = t.slice(0, MAX_CHARACTERS-4);
      t = `"${t}..."`;
    }
    return t;
  };

  const handleOnChangeVisibility = (visible) => {
    setShowTextPopUp(visible);
  };

  const handleCopyLink = () => {
    var temp = document.createElement("textarea");
    temp.value = textItem.source;
    document.body.appendChild(temp);
    temp.select();
    navigator.clipboard.writeText(temp.value);
    document.body.removeChild(temp);

    var tooltip = document.getElementById(textItem._id);
    tooltip.innerHTML = "Copied!";
  };

  return (
    <>
      <div className="TextBox">
        <div className="text" onClick={() => setShowTextPopUp(true)}>
          {formatText(textItem.text)}
        </div>
        <div className="divider"/>
        <div className="card-footer">
          <div className="TextBox-source-tooltip">
            <div className="TextBox-source" onClick={handleCopyLink}>
              <span id={textItem._id} className="TextBox-source-tooltip-text">Click to copy link</span>
              <ChainIcon className="TextBox-source-icon"/>
              <div className="TextBox-source-text">{trimLongText(textItem.source, 30)}</div>
            </div>
          </div>
          <div onClick={() => handleDelete(textItem._id)} className="delete">
            <TrashCanIcon/>
          </div>
        </div>
      </div>

      {/* Text Pop Up */}
      { showTextPopUp && <TextBoxPopUp onChangeVisibility={handleOnChangeVisibility} text={textItem.text} source={textItem.source}/>}
    </>
  );
}

TextBox.propTypes = {
  textItem: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    source: PropTypes.string.isRequired,
    creationDate: PropTypes.number.isRequired
  }),
  onDelete: PropTypes.func.isRequired
};

export default TextBox;