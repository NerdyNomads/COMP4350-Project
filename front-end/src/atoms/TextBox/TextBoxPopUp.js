import React from "react";
import PropTypes from "prop-types";

import { ChainIconSmallIcon, ExitIcon } from "../icons";

import "./TextBoxPopUp.css";

const MAX_CHAR_FOR_SOURCE_TO_SHOW = 50;

function TextBoxPopUp( {text, source, onChangeVisibility} ) {

  const handleBackgroundClick = (e) => {
    // clicked outside of modal
    if(e.target.className == "TextBoxPopUp-background") {
      onChangeVisibility(false);
    }
  };

  const handleExitClick = () => {
    onChangeVisibility(false);
  };

  const handleOnSourceClick = () => {
    alert(`Link copied: ${source}`);

    //Create a temporary element to be able to copy the source
    var temp = document.createElement("textarea");
    temp.value = source;
    document.body.appendChild(temp);
    temp.select();
    document.execCommand("copy");
    document.body.removeChild(temp);
  };

  const renderSource = (srcText) => {
    return srcText.substring(0, MAX_CHAR_FOR_SOURCE_TO_SHOW) + "...";
  };

  return (
    <div className="TextBoxPopUp-background" onClick={handleBackgroundClick}>
      <div className="TextBoxPopUp">
        <div className="TextBoxPopUp-exit" onClick={handleExitClick}>
          <ExitIcon/>
        </div>
        <div className="TextBoxPopUp-source" onClick={handleOnSourceClick}>
          <div className="TextBoxPopUp-source-icon">
            <ChainIconSmallIcon/>
          </div>
          <div className="TextBoxPopUp-source-text">
            <a className="TextBoxPopUp-source-text-a">{renderSource(source)}</a>
          </div>
        </div>
        <div className="TextBoxPopUp-text">
          {text}
        </div>
      </div>
    </div>
  );
}

TextBoxPopUp.propTypes = {
  text: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired,
  onChangeVisibility: PropTypes.func.isRequired,
};


export default TextBoxPopUp;