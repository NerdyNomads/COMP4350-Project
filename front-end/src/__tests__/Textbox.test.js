/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import TextBox from "../atoms/TextBox";
import axios from "axios";

jest.mock("axios");

let props;
const longText =
  "An application programming interface (API) is a connection between computers or between computer programs. It is a type of software interface, offering a service to other pieces of software. A document or standard that describes how to build or use such a connection or interface is called an API specification.";

beforeEach(() => {
  props = {
    textItem: {
      _id: "1",
      creationDate: 0,
      source: "",
      text: "Test Text",
    },
    onDelete: jest.fn(),
  };
});

test("Should match the snapshot.", () => {
  const wrapper = shallow(<TextBox {...props} />);
  expect(wrapper.html()).toMatchSnapshot();
});

test("Logout button should not be visible if a user is not logged in.", () => {
  const wrapper = shallow(<TextBox {...props} />);
  expect(wrapper.find("button").length).toEqual(0);
});

test("Long text being cut off", () => {
  props.textItem.text = longText;
  const wrapper = shallow(<TextBox {...props} />);
  expect(wrapper.text()).not.toContain(longText);
  expect(wrapper.text()).toContain("...");
});

test("Short text not being cut off", () => {
  const wrapper = shallow(<TextBox {...props} />);
  expect(wrapper.text()).toContain(props.textItem.text);
  expect(wrapper.text()).not.toContain("...");
});

test("Deletion", async () => {
  const wrapper = shallow(<TextBox {...props} />);
  // wrapper.find(".delete").to.have.length(1); //simulate("click");
  await wrapper.find("div.delete").simulate("click");
  () => expect(props.onDelete).toHaveBeenCalledTimes(1);
});
