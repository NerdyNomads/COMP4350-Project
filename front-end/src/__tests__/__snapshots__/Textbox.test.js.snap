// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"TextBox\\"><div class=\\"text\\">Test Text</div><div class=\\"divider\\"></div><div class=\\"card-footer\\"><div class=\\"source\\" target=\\"_blank\\" rel=\\"noreferrer\\"><img alt=\\"link\\" src=\\"/assets/images/chain.svg\\" class=\\"icon\\"/></div><div id=\\"1\\"></div><div class=\\"delete\\"><img alt=\\"trash can\\" src=\\"/assets/images/trash_can.svg\\" class=\\"icon\\"/></div></div></div>"`;
